picoCTF-web/api/events.py

```diff
@@ -0,0 +1,65 @@
+"""
+Module for dealing with events.
+
+An event, typically a competition, has associated eligibility criteria.
+A team is initially eligible for any events for which its founding member
+is eligibile.
+
+When a new member joins the team, the team's eligibility for a given event
+will be revoked if the new member does not fit its criteria.
+(However, by default, members are prevented from joining a team if doing
+so would cause the team to lose any existing eligibility status.)
+
+Events can also have certain metadata, such as a sponsor, logo, etc.
+"""
+
+import api
+
+
+def get_all_events():
+    """Return a list of all events in the database."""
+    db = api.db.get_conn()
+    events = db.events.find({}, {'_id': False})
+    if not events:
+        return []
+    else:
+        return list(events)
+
+
+def get_event(eid):
+    """Return an event from the database, or None if it does not exist."""
+    db = api.db.get_conn()
+    return db.events.find_one({'eid': eid}, {'_id': False})
+
+
+def add_event(name, eligibility_conditions="", sponsor=None, logo=None):
+    """
+    Add an event to the database.
+
+    Args:
+        name (str): name of the event
+        eligibility_conditions (str): mongodb query to find eligible users
+        sponsor (str): optional, sponsor of the event
+        logo (str): optional, URL of a logo image for the event
+
+    Returns:
+        ID of the newly created event
+    """
+    db = api.db.get_conn()
+    eid = api.common.token()
```

Comment:
Can we key events not on a uuid but their name instead (with some quick error if name exists already)? This would make all of the eligibility checks a lot easier when we can filter db.teams directly by names of events in eligibilities. That way we can pass along the name directly for scoreboards (e.g. `global`, `us_student`) etc